tests/lib/prepare.sh

```diff
@@ -505,15 +515,30 @@ uc20_build_initramfs_kernel_snap() {
         # modify the-tool to verify that our version is used when booting - this
         # is verified in the tests/core/basic20 spread test
         sed -i -e 's/set -e/set -ex/' "$skeletondir/main/usr/lib/the-tool"
-        echo "" >> "$skeletondir/main/usr/lib/the-tool"
-        echo "if test -d /run/mnt/data/system-data; then touch /run/mnt/data/system-data/the-tool-ran; fi" >> \
-            "$skeletondir/main/usr/lib/the-tool"
+        # also save the time before snap-bootstrap runs
+        sed -i -e "s@/usr/lib/snapd/snap-bootstrap@beforeDate=\$(date --utc \'+%s\'); /usr/lib/snapd/snap-bootstrap@"  "$skeletondir/main/usr/lib/the-tool"
+        {
+            echo "" 
```

Comment:
(nitpick^2) we could use 
```
cat >> "$skeletondir/main/usr/lib/the-tool" <<'EOF'
if test -d /run/mnt/data/system-data; then touch /run/mnt/data/system-data/the-tool-ran; fi
# also copy the time for the clock-epoch to system-data, this is 
# used by a specific test but doesn't hurt anything to do this for 
# all tests
mode=$(grep -Eo 'snapd_recovery_mode=([a-z]+)' /proc/cmdline)
echo "mode=${mode##snapd_recovery_mode=}
stat -c '%Y' /usr/lib/clock-epoch >> /run/mnt/ubuntu-seed/${mode}-clock-epoch
echo "$beforeDate" > /run/mnt/ubuntu-seed/${mode}-before-snap-bootstrap-date
date --utc '+%s' > /run/mnt/ubuntu-seed/${mode}-after-snap-bootstrap-date
EOF
```
here for easier readability maybe? Note that with the quotes around 'EOF' there should be no need for quoting, nothing will be expanded by the shell then (I assume this is what we want at this point?). But if we do it, doing it in a followup is totally fine, should not block this PR).

Reply:
Yeah using a heredoc would probably be more readable, I'll do this in a follow-up

Reply:
it seems looking back at this PR I never did do this in a followup /o\ will add a TODO for myself to come back and actually do this now, 9 months later

Reply:
@anonymouse64 I'm making this change in another PR where I'm also trying to have bit fewer top-level test files in ubuntu-seed